client/telemetry/src/layer.rs

```diff
@@ -0,0 +1,112 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use futures::channel::mpsc;
+use parking_lot::Mutex;
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::sync::Arc;
+use tracing::{Event, Subscriber};
+use tracing_subscriber::{layer::Context, registry::LookupSpan, Layer};
+
+pub const TELEMETRY_LOG_SPAN: &str = "telemetry-logger";
+
+pub struct TelemetryLayer(Senders);
+
+impl TelemetryLayer {
+	pub fn new() -> Self {
+		Self(Default::default())
```

Comment:
You can also just derive Default and remove this function

Reply:
right! thanks!!